File: schemas/multi-language-i18n/drizzle/translation_status_history.ts
// translation_status_history: Audit log of translation status changes.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp, index } from "drizzle-orm/pg-core";
import { users } from "../../auth-rbac/drizzle/users";

export const translationStatusHistory = pgTable(
  "translation_status_history",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    translationType: text("translation_type").notNull(),
    translationId: uuid("translation_id").notNull(),
    fromStatus: text("from_status"),
    toStatus: text("to_status").notNull(),
    changedBy: uuid("changed_by").references(() => users.id, {
      onDelete: "set null",
    }),
    comment: text("comment"),
    createdAt: timestamp("created_at", { withTimezone: true })
      .notNull()
      .defaultNow(),
  },
  (table) => [
    index("idx_translation_status_history_translation_type_translation_id").on(
      table.translationType,
      table.translationId
    ),
    index("idx_translation_status_history_changed_by").on(table.changedBy),
  ]
);

File: schemas/multi-language-i18n/spacetimedb/translation_status_history.rs
// translation_status_history: Append-only log of translation status changes.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

// Composite index: (translation_type, translation_id) — full status history for a translation

#[spacetimedb::table(name = translation_status_history, public)]
pub struct TranslationStatusHistory {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub translation_type: String,

    pub translation_id: String,
    pub from_status: Option<String>,
    pub to_status: String,

    #[index(btree)]
    pub changed_by: Option<String>, // FK → users.id (set null)

    pub comment: Option<String>,
    pub created_at: Timestamp,
}
